### PR TITLE
Use regex for bearer token validation

### PR DESCRIPTION
**Pull request by divergentdave**
This implements the suggestion in https://github.com/divviup/janus/pull/1985/files#r1336411760.

### CODE REVIEW

File: core/src/auth_tokens.rs [hunk 1]
```diff
@@ -201,36 +202,15 @@ impl BearerToken {
...
-        for c in &mut iter {
-            match c {
-                '=' => {}
-                _ => {
-                    return Err(anyhow::anyhow!(
-                        "bearer token may only contain '=' at the end"
-                    ))
-                }
-            }
+        static REGEX: OnceLock<Regex> = OnceLock::new();
```
Comment (inahga) [inline core/src/auth_tokens.rs:205]: Neat-o, `OnceLock` was stabilized in 1.70, which is now our MSRV. We can probably drop `lazy_static`, with a bit of effort.

**Review by inahga — APPROVED** [review]
I'm partial to either approach. This regex turns out to be not unreadable.

File: core/src/auth_tokens.rs (hunk 1 above)
Comment (divergentdave) [reply core/src/auth_tokens.rs:205]: Already done in #1852 :grin:

Comment (inahga) [reply core/src/auth_tokens.rs:205]: Aha, while I was on vacation :laughing:

**Pull request merged**